addons/portal_rating/static/src/chatter/frontend/message_model_patch.js

```diff
@@ -0,0 +1,16 @@
+import { Message } from "@mail/core/common/message_model";
+import { patch } from "@web/core/utils/patch";
+
+/** @type {import("models").Message} */
+const messagePatch = {
+    async remove({ removeFromThread = false } = {}) {
+        const data = await super.remove(...arguments);
+        if (this.thread && removeFromThread) {
+            this.thread.messages.forEach((message) => {
+                message.rating_stats = this.thread.rating_stats;
+            });
```

Comment:
The rating_stats of each message should be updated when an updated rating_stats value is received in the thread data (`/mail/message/update_content`).

Note: All messages fetched via the portal route have the thread's rating_stats (the same value for all messages).